synapse/lib/auth.py

```diff
@@ -1332,7 +1333,25 @@ async def tryPasswd(self, passwd, nexs=True):
             return False
 
         if isinstance(shadow, dict):
-            return await s_passwd.checkShadowV2(passwd=passwd, shadow=shadow)
+            result = await s_passwd.checkShadowV2(passwd=passwd, shadow=shadow)
+            if self.auth.policy and (attempts := self.auth.policy.get('attempts')) is not None:
+                if result:
+                    await self.auth.setUserInfo(self.iden, 'policy:attempts', 0)
+                    return True
+
+                valu = self.info.get('policy:attempts', 0) + 1
+                await self.auth.setUserInfo(self.iden, 'policy:attempts', valu)
+
+                if valu >= attempts:
+                    await self.auth.nexsroot.cell.setUserLocked(self.iden, True)
```

Comment:
We should reset `policy:attempts` to 0 within `setUserLocked()` so they get a clean slate once they're unlocked.